enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/ast/NodeProperty.scala

```diff
@@ -19,20 +19,22 @@
  */
 package org.neo4j.cypher.internal.compatibility.v3_4.runtime.ast
 
-case class NodeProperty(offset: Int, propToken: Int, name: String) extends RuntimeExpression {
+import org.neo4j.cypher.internal.v3_4.expressions.Property
+
+case class NodeProperty(offset: Int, propToken: Int, name: String)(prop: Property) extends RuntimeProperty(prop) {
```

Comment:
Why can't we have `..., name: String, prop: Property)`? Applies to everything that extends `RuntimeProperty`

Reply:
It leads to infinite recursion in the top down tree rewriter (I tried...)